app/services/github.js

```diff
@@ -0,0 +1,46 @@
+import {inject} from '@ember/service';
+import AjaxService from 'ember-ajax/services/ajax';
+import {generate} from '../graphql/github';
+
+export default AjaxService.extend({
+  userSettings: inject(),
+  githubUrl: 'https://api.github.com/graphql',
+
+  init() {
+    this._super(...arguments);
+    this.headers = {
+      Authorization: "bearer " + this.userSettings.tokens.get('github_com')
+    }
+  },
+
+  request(data) {
+    return this._super(this.githubUrl, {
+      method: 'post',
+      data: JSON.stringify(data),
+      processData: false,
+      headers: {
+        Authorization: "bearer " + this.userSettings.tokens['github_com']
+      },
+      success() {
+        return {};
+      }
+    })
+  },
+
+  tasks(context) {
+    const transform = (response) => { 
+      return response.data.search.nodes.map((node) => {
+        node.isPullRequest = node.type == 'PullRequest';
+        return node;
+      })
+    }
+    const query = generate('tasks', context);
+    const request = this.request({query: query})
```

Comment:
If you don't want to use GraphQL library, atleast send the variable, there's no need to parse or do variables replacing yourself. 

IIRC, variables is sent as `variables` key as JSON.

Reply:
It doesn't parse. It transform to a github search query. I don't know if github support array :). The `query` argument only support string https://developer.github.com/v4/query/#connections

Reply:
@blazeu We still need to transform, even though we use library.

Reply:
However, I do agree that i should not interpolate (Replacing) the query. and utilize variable instiead.